app/src/main/java/com/github/fribourgsdp/radio/TimerProgressBarHandler.kt

```diff
@@ -0,0 +1,42 @@
+package com.github.fribourgsdp.radio
+
+import android.view.View
+import android.widget.ProgressBar
+import java.util.*
+
+/**
+ * An instance of a handler to link a [progress bar][ProgressBar] and a [timer][Timer] together.
+ *
+ * NB:
+ *
+ * The parameter [timerListener][Timer.Listener] is responsible to update the [progress bar][progressBar]
+ * since the update needs to be done on the main ui thread.
+ *
+ * To know what is the current time, we can give a function currentTimeInMillis. By default, it's [System.currentTimeMillis]
+ *
+ * @property timer the [timer] of the [handler][TimerProgressBarHandler].
+ * @property progressBar the [progress bar][progressBar] of the [handler][TimerProgressBarHandler].
+ */
+class TimerProgressBarHandler(val timer: Timer, val progressBar: ProgressBar, timerListener: Timer.Listener, private val currentTimeInMillis: () -> Long = System::currentTimeMillis): Timer.DeadlineHandler {
+
+    init {
+        timer.setListener(500L, timerListener)
```

Comment:
Could you extract this 500L so that we know directly what it corresponds to?

Reply:
OK you're right it's better 👍🏻